provisioning_client/src/prov_device_ll_client.c

```diff
@@ -204,7 +214,11 @@ static void on_transport_error(PROV_DEVICE_TRANSPORT_ERROR transport_error, void
 static PROV_DEVICE_TRANSPORT_STATUS retrieve_status_type(const char* prov_status)
 {
     PROV_DEVICE_TRANSPORT_STATUS result;
-    if (prov_status == NULL || strcmp(prov_status, PROV_UNASSIGNED_STATUS) == 0)
+    if (prov_status == NULL)
+    {
+        result = PROV_DEVICE_TRANSPORT_STATUS_ERROR;
```

Comment:
>PROV_DEVICE_TRANSPORT_STATUS_ERROR [](start = 17, length = 34)

In a normal world, we would do a null check right after we obtained the JSON in the calling function. This isn't a 'request failed' scenario. So, I would treat this as a local issue and failfast in the calling function and remove this NULL check here.